nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/ConfigProviderResolverSync.java

```diff
@@ -96,4 +95,17 @@ public void registerConfig(Config config, ClassLoader classLoader) {
     public void releaseConfig(Config config) {
         await().releaseConfig(config);
     }
+
+
+    private ConfigProviderResolver await() {
+        LOG.log(Level.WARNING, "Payara Microprofile Config requested too early, the HK2 service is not initialized yet."
+            + " Waiting until it will be active.");
+        while (true) {
+            final ConfigProviderResolver resolver = instance();
+            if (resolver != null && resolver != this) {
+                return resolver;
+            }
+            Thread.yield();
```

Comment:
.yield() is a hint, which means that control isn't necessarily  passed on to another Thread.  A sleep with very small amount should be used.

Reply:
I used it on Linux+Intel/AMD and on Solaris/SPARC, even old versions, and it always worked. Also it is only few blocks after startup, so it does not slow down the Payara Startup. Sleep is unnecessary overkill, because I don't need to sleep running thread, I only tell the scheduler "at this moment I have nothing to do, but if other threads have some work, they can".
https://www.geeksforgeeks.org/java-concurrency-yield-sleep-and-join-methods/

Reply:
`instance()` also has synchronized block, so there's another yield point there, so I believe this might be correct.

But still, I trust `CountDownLatch` more, but might be overkill for 1 producer to 1 consumer scenario.